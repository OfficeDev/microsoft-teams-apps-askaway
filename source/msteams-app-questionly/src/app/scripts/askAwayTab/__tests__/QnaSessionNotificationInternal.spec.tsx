/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, Button, Form, Input, TextArea } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import QnaSessionNotificationInternal from '../popups/QnaSessionNotificationInternal';

configure({ adapter: new Adapter() });

describe('QnaSessionNotificationInternal Component', () => {
    const onSubmitSession = jest.fn();
    const searchParams = {} as URLSearchParams;

    it('should match the snapshot', () => {
        const wrapper = shallow(<QnaSessionNotificationInternal onSubmitSession={onSubmitSession} searchParams={searchParams} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<QnaSessionNotificationInternal onSubmitSession={onSubmitSession} searchParams={searchParams} />);

        expect(component.find(Flex)).toHaveLength(2);
        expect(component.find(Text)).toHaveLength(3);
        expect(component.find(Button)).toHaveLength(1);
    });
});
