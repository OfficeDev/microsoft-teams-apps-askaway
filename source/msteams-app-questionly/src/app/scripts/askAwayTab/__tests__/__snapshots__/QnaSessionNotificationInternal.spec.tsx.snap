// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QnaSessionNotificationInternal Component should match the snapshot 1`] = `
<Flex
  column={true}
  gap="gap.medium"
  padding="padding.medium"
>
  <Text
    as="span"
    content="popups.notificationTitle"
  />
  <Text
    as="span"
    content="some-value"
    size="medium"
    weight="bold"
  />
  <Text
    as="span"
    content="popups.notificationMessage"
  />
  <FlexItem
    align="end"
  >
    <Button
      accessibility={[Function]}
      as="button"
      content="popups.notificationButton"
      onClick={[Function]}
      primary={true}
      size="small"
      type="submit"
    />
  </FlexItem>
</Flex>
`;
